From bf2fe41616c54712f78ea0ffd173f3614eb20190 Mon Sep 17 00:00:00 2001
From: Pino Toscano <toscano.pino@tiscali.it>
Date: Sun, 21 Nov 2021 09:25:30 +0100
Subject: [PATCH] QStorageInfo: limit Linux-only code with Q_OS_LINUX

This fixes a build problem (use of PATH_MAX) by eliminating a
Linux-specific condition that can't happen on HURD anyway.

Change-Id: I5dcaf104a60b7850b8af3964fc4cd02ab24acd7b
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 5558eb4edb62ddc920fd5f4171e732fb7cbe53a7)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/io/qstorageinfo_unix.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/corelib/io/qstorageinfo_unix.cpp b/src/corelib/io/qstorageinfo_unix.cpp
index 566ffe46bc..2b635f63bb 100644
--- a/src/corelib/io/qstorageinfo_unix.cpp
+++ b/src/corelib/io/qstorageinfo_unix.cpp
@@ -568,6 +568,7 @@ inline QByteArray QStorageIterator::fileSystemType() const
 
 inline QByteArray QStorageIterator::device() const
 {
+#ifdef Q_OS_LINUX
     // check that the device exists
     if (mnt.mnt_fsname[0] == '/' && access(mnt.mnt_fsname, F_OK) != 0) {
         // It doesn't, so let's try to resolve the dev_t from /dev/block.
@@ -583,6 +584,7 @@ inline QByteArray QStorageIterator::device() const
             return dev;
         }
     }
+#endif
     return QByteArray(mnt.mnt_fsname);
 }
 
-- 
2.34.1

